tests/test_utils.py

```diff
@@ -0,0 +1,54 @@
+# This file is part of ap_association.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import tempfile
+import unittest
+
+import pandas as pd
+
+from lsst.ap.association.utils import readSchemaFromApdb, make_empty_catalog, convertTableToSdmSchema
+import lsst.dax.apdb as daxApdb
+
+
+class TestUtils(unittest.TestCase):
+
+    def test_make_empty_catalog(self):
+        """Check that an empty catalog has the correct format.
+        """
+        apdb_config = daxApdb.ApdbSql.init_database(db_url="sqlite://")
```

Comment:
IIRC, in-memory APDBs will be removed in a future version. But I think @andy-slac would need to deprecate them first?

Reply:
In-memory SQLite support is already removed since DM-43414. It might work for this particular unit test, but you still should not use it.

Reply:
In that case, I recommend using `NamedTemporaryFile` [as a context manager](https://docs.python.org/3.11/library/tempfile.html#tempfile.NamedTemporaryFile) instead of using `addCleanup` (which is more appropriate for stuff created in `setUp`):
```suggestion
        with tempfile.NamedTemporaryFile as apdb_file:
            apdb_config = daxApdb.ApdbSql.init_database(db_url="sqlite://" + apdb_file)
            apdb = daxApdb.Apdb.from_config(apdb_config)
```

Reply:
I was struggling to get your suggestion to work, and realized I only need the schema here, not the full APDB. I will use `readSdmSchemaFile` instead of `readSchemaFromApdb` to get the schema.